src/user_input_compiler.rs

```diff
@@ -0,0 +1,258 @@
+//! # user_input_compiler
+//!
+//! Parses Input with this Grammer after trimming the Input and ignoring Whitespaces:
+//! 
+//! Command         := "/" CommandWord
+//! CommandWord     := Status | Subscribe | Cancel | InvalidSyntax
+//! Status          := "status" | "crowd"
+//! Subscribe       := "subscribe" SensorSelector Duration
+//! SensorSelector  := SensorString OptionalNum
+//! SensorString    := "account_balance" | "barometer" | "beverage_supply" | "door_locked" | "humidity" | "network_connections" | "power_consumption" | "temperature" | "total_member_count" | "radiation.alpha" | "radiation.beta_gamma" | "radiation.beta" | "radiation.gamma" | "people_now_present" | "wind"
+//! OptionalInteger := Integer | ɛ
+//! Integer         := [0-9]*
+//! Duration        := Real TimeSuffix
+//! TimeSuffix      := "m" | "min" | "h" | "d"
+//! Real            := Integer "." Integer | Integer
+//! Cancel          := "cancel"
+//! InvalidSyntax   := *
```

Comment:
I'm not sure whether parse errors should be part of the grammar. I think the lexer should be able to handle that.